src/hooks/useErc20.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react'
+
+import { TokenErc20 } from '@gnosis.pm/dex-js'
+
+import { Network } from 'types'
+
+import { buildErc20Key, saveSingleErc20 } from 'reducers-actions/erc20'
+
+import useGlobalState from 'hooks/useGlobalState'
+
+import { getErc20Info } from 'services/helpers'
+
+import { web3, erc20Api } from 'apps/explorer/api'
+
+import { ExplorerAppState } from 'apps/explorer/state'
+
+export function useErc20(
+  address?: string,
+  networkId?: Network,
+): { isLoading: boolean; error?: string; value?: TokenErc20 } {
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState('')
+
+  const [{ erc20s }, dispatch] = useGlobalState<ExplorerAppState>()
+
+  // Initial state from global state
+  const [erc20, setErc20] = useState<TokenErc20 | undefined>(
```

Comment:
does `undefined` need to be explicit here?

Reply:
Yes, otherwise TS complains